spec/plugin_spec.rb

```diff
@@ -30,10 +30,12 @@ def register_css(arg)
 describe OAuth2BasicAuthenticator do
   context 'after_authenticate' do
     let(:user) { Fabricate(:user) }
-    let(:authenticator) { OAuth2BasicAuthenticator.new('oauth2_basic') }
+    let(:authenticator) { OAuth2BasicAuthenticator.new() }
 
     let(:auth) do
-      { 'credentials' => { 'token': 'token' },
+      { 'provider' => 'oauth2_basic',
```

Comment:
As mentioned above, this should be changed to 
```suggestion
      OmniAuth::AuthHash.new { 'provider' => 'oauth2_basic',
```
Then the tests will match production behaviour

Reply:
Yup also fair. I've made that change.